src/libraries/System.Collections/src/System/Collections/Generic/SortedList.cs

```diff
@@ -402,18 +402,21 @@ bool ICollection.IsSynchronized
         // Removes all entries from this sorted list.
         public void Clear()
         {
-            // clear does not change the capacity
-            version++;
-            // Don't need to doc this but we clear the elements so that the gc can reclaim the references.
-            if (RuntimeHelpers.IsReferenceOrContainsReferences<TKey>())
-            {
-                Array.Clear(keys, 0, _size);
-            }
-            if (RuntimeHelpers.IsReferenceOrContainsReferences<TValue>())
+            if (_size != 0)
```

Comment:
Why add this check?

Reply:
If the size is already 0, then there are no elements to clear from the internal list, so we can avoid those calls to `RuntimeHelpers` and can exit directly.

Reply:
`RuntimeHelpers.IsReferenceOrContainsReferences` isn't a "call". It will be treated as a **constant** by JIT.